File: pulsar-ql-server/src/main/java/ai/platon/pulsar/ql/start/H2DbConsole.java
package ai.platon.pulsar.ql.start;

import ai.platon.pulsar.PulsarEnv;
import ai.platon.pulsar.common.SystemKt;
import org.h2.tools.Console;
import org.springframework.boot.CommandLineRunner;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.ApplicationContext;
import org.springframework.context.ConfigurableApplicationContext;
import org.springframework.context.annotation.Bean;

import java.sql.SQLException;
import java.util.Arrays;

/**
 * Server start port
 * */
@SpringBootApplication
public class H2DbConsole {
    private static PulsarEnv env = PulsarEnv.Companion.getOrCreate();
    private static ApplicationContext applicationContext;

    @Bean
    public CommandLineRunner commandLineRunner(ApplicationContext ctx) {
        return args -> {
            String[] beanNames = ctx.getBeanDefinitionNames();
            Arrays.sort(beanNames);
            System.out.println("Bean definition names: ");
            for (String beanName : beanNames) {
                System.out.println(beanName);
            }

            new Console().runTool(args);
        };
    }

    public static void main(String[] args) {
        SystemKt.setPropertyIfAbsent("h2.sessionFactory", ai.platon.pulsar.ql.h2.H2SessionFactory.class.getName());
        SpringApplication.run(H2DbConsole.class, args);
    }
}

File: pulsar-ql-server/src/main/kotlin/ai/platon/pulsar/ql/h2/start/H2DbConsole.kt
package ai.platon.pulsar.ql.h2.start

import ai.platon.pulsar.PulsarEnv
import ai.platon.pulsar.common.config.CapabilityTypes
import ai.platon.pulsar.common.setPropertyIfAbsent
import com.google.common.collect.Lists
import org.h2.tools.Console
import org.slf4j.LoggerFactory
import org.springframework.boot.CommandLineRunner
import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.autoconfigure.mongo.MongoAutoConfiguration
import org.springframework.boot.autoconfigure.mongo.embedded.EmbeddedMongoAutoConfiguration
import org.springframework.context.ApplicationContext
import org.springframework.context.annotation.Bean

/**
 * Server start port
 */
@SpringBootApplication(exclude = [MongoAutoConfiguration::class, EmbeddedMongoAutoConfiguration::class])
open class H2DbConsole {
    val log = LoggerFactory.getLogger(H2DbConsole::class.java)

    @Bean
    open fun commandLineRunner(ctx: ApplicationContext): CommandLineRunner {
        return CommandLineRunner { args ->
            if (log.isInfoEnabled) {
                val beans = ctx.beanDefinitionNames.sorted()
                val s = Lists.partition(beans, 10).joinToString("\n\t") { it.joinToString { it } }
                log.info("Defined beans: {}", s)
            }
            Console().runTool(*args)
        }
    }

    companion object {
        private val env = PulsarEnv.getOrCreate()

        @JvmStatic
        fun main(args: Array<String>) {
            setPropertyIfAbsent("h2.sessionFactory", ai.platon.pulsar.ql.h2.H2SessionFactory::class.java.name)
            // setPropertyIfAbsent(CapabilityTypes.PROXY_ENABLE_INTERNAL_SERVER, "true")

            SpringApplication.run(H2DbConsole::class.java, *args)
        }
    }
}
